build/tools/Automation/DotNet/DotNet.cs

```diff
@@ -0,0 +1,236 @@
+                    dotNetYamlConstant.Constants[dotNetConstantKey] = version;
+
+                    // add sdk to versionsToBuild.txt
+                    UpdateVersionsToBuildTxt(platformConstant);
+                }
+                else
+                {
+                    Constant dotNetYamlConstant = dotnetYamlConstants[Constants.DotNetRuntimeKey];
+                    dotNetYamlConstant.Constants[dotNetConstantKey] = version;
+
+                    // store SHAs for net-core and aspnet-core
+                    dotNetYamlConstant.Constants[$"{dotNetConstantKey}-sha"] = sha;
+                }
+            }
+
+            var serializer = new SerializerBuilder()
+                .WithNamingConvention(UnderscoredNamingConvention.Instance)
+                .Build();
+
+            var stringResult = serializer.Serialize(yamlConstants);
+            File.WriteAllText(Constants.ConstantsYaml, stringResult);
+        }
+
+        private static void UpdateVersionsToBuildTxt(PlatformConstant platformConstant)
+        {
+            List<string> versionsToBuildTxtFiles = new List<string>() {
+                    "platforms/dotnet/versions/bullseye/versionsToBuild.txt",
+                    "platforms/dotnet/versions/buster/versionsToBuild.txt",
+                    "platforms/dotnet/versions/focal-scm/versionsToBuild.txt",
+                    "platforms/dotnet/versions/stretch/versionsToBuild.txt",
+            };
+            foreach (string versionsToBuildTxtFile in versionsToBuildTxtFiles)
+            {
+                string line = $"\n{platformConstant.Version}, {platformConstant.Sha},";
+                File.AppendAllText(versionsToBuildTxtFile, line);
+
+                // sort
+                Console.WriteLine($"Updating {versionsToBuildTxtFile}...");
+                var contents = File.ReadAllLines(versionsToBuildTxtFile);
+                Array.Sort(contents);
+                File.WriteAllLines(versionsToBuildTxtFile, contents);
+            }
+        }
+
+        private static Dictionary<string, Constant> GetYamlDotNetConstants(List<Constant> yamlContents)
+        {
+            Dictionary<string, Constant> dotNetConstants = new Dictionary<string, Constant>();
+            foreach (var constant in yamlContents)
+            {
+                if (constant.Name == Constants.DotNetSdkKey ||
+                    constant.Name == Constants.DotNetRuntimeKey)
+                {
+                    dotNetConstants.Add(constant.Name, constant);
+                }
+            }
+            return dotNetConstants;
+        }
+
+        private static string GenerateDotNetConstantKey(PlatformConstant platformConstant)
+        {
+            string[] splitVersion = platformConstant.Version.Split('.');
+            string majorVersion = splitVersion[0];
+            string minorVersion = splitVersion[1];
+            string majorMinor = majorVersion + minorVersion;
```

Comment:
I wouldn't use `majorVersion + minorVersion` as the key because we could easily be bitten in the scenario where we have `11.0` and `1.10` and both keys turn out to be `110`. We should include as much information as we can into the key to make it a unique string compared to any other version release.

Reply:
I feel this is probably something we should prioritize for now and get it out of the way to not leave us vulnerable for missing releases.

Reply:
Updated: https://github.com/microsoft/Oryx/pull/1560/commits/fdead7e4bf24cd498f6876ae9916cd12cbeef86e#diff-947811409334f4664b049add37fc8940c9c822b2cd039cd24b3d9a7cd410716dR200

One way this falls short is if we have .NET version 100+ 
I considered appending the releaseDate if the major version is above 9 but it seems unnecessary.